Adding links to the landing page (#2460)
dafny-lang/dafny: docs/index.md

---

> title: The Dafny Language and Software Verification System

Author: By "Software Verification System" I meant the tools used to check/compile etc. Dafny programs. The project is not just language design -- it is tool development for that language.

> title: The Dafny Language and Software Verification System

Reviewer: Fair enough, but IMO the main `dafny.org` front page should be primarily focussed on Dafny users rather than Dafny tool developers. We can have subpages dedicated to developing Dafny itself.

We could have something like "The Dafny Verified Programming Language" if we wanted to emphasize what makes Dafny unique.

> title: The Dafny Language and Software Verification System

Author: I agree fully that dafny.org is for users (the dafny GitHub project and wiki are for developers). But the users are using the tools, not just learning the language.

# Feel free to add content and custom Front Matter to this file.
# To modify the layout, see https://jekyllrb.com/docs/themes/#overriding-theme-defaults

layout: page
---
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script type="text/x-mathjax-config">
        MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ["\\(","\\)"]], displayMath: [ ["$$","$$"], ["\\[","\\]"] ]
        }});

> </script>

Reviewer 2: Do we foresee using any of those?

Author: MathJax is certainly used. I have a separate Issue to cleanup other html that is no longer used.

Reviewer 2: In the index?

Author: In all files under docs

<link rel="stylesheet" href="assets/main.css">

**Quick Links:**
- [Dafny Reference Manual and User Guide](DafnyRef/DafnyRef)
- [Index to Dafny Resources for Users](toc)

> - [Dafny GitHub project (for developers of the Dafny tools themselves)](https://github.com/dafny-lang/dafny)

Reviewer 2: This may be  a good spot to link to the VS code extension.

Author: Added the link

> - [Index to Dafny Resources for Users](toc) ([installation](https://github.com/dafny-lang/dafny/wiki/INSTALL)) ([Dafny for VSCode](https://marketplace.visualstudio.com/items?itemName=dafny-lang.ide-vscode))

Reviewer 2: ```suggestion
- [Index to Dafny Resources for Users](toc) 
- Install Dafny using the Visual Studio Code [extension](https://marketplace.visualstudio.com/items?itemName=dafny-lang.ide-vscode)
```

Author: If we are going to include installation for VSCode here, why not a link to the installation page itself? -- not everyone uses VSCode.

Reviewer 2: Very true, but in my mind, this page doesn't target everyone, but people new to Dafny, who just want to get started. I don't want to overwhelm them with a plethora of options.

Author: Right -- so some new person comes to the page -- Linux or Mac user - and wonders how to install. Really? I have to use VSCode? I've never used it before! That's Microsoft! How about a command-line? or emacs?

Reviewer 2: I'm not saying that those user don't exist, but I trust them to find what they're looking for themselves.

![Dafny Banner](./banner.png "Dafny")

Dafny is a **verification-ready programming language**.
As you type in your program, Dafny's verifier constantly looks over your shoulder, flags any errors, shows you counterexamples, and congratulates you when your code matches your specifications.

> When you're done, Dafny can **compile your code to C#, Java, JavaScript, Python, C++ or Go** (more to come!), so it can integrate with your existing workflow.

Reviewer 2: ```suggestion
When you're done, Dafny can **compile your code to C#, Java, JavaScript, C++ or Go** (more to come!), so it can integrate with your existing workflow.
```
The current plan is to announce Python once it's ready with Dafny 4.

---
# Feel free to add content and custom Front Matter to this file.
# To modify the layout, see https://jekyllrb.com/docs/themes/#overriding-theme-defaults

layout: page
---
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script type="text/x-mathjax-config">
        MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ["\\(","\\)"]], displayMath: [ ["$$","$$"], ["\\[","\\]"] ]
        }});
</script>

<link rel="stylesheet" href="assets/main.css">

> <font size="+4"><p style="text-align: center;">The Dafny Language and Software Verification System</p></font>

Reviewer 3: `<font>` has been deprecated for years; let's not use that.  Should this be an `<h1>`?
Ideally though it would be best to stick with Markdown; we can give the title an ID and style it with CSS

Author: Corrected. (the html came from a page-generating tool that we are no longer using)

**Quick Links:**
- [Dafny Reference Manual and User Guide](DafnyRef/DafnyRef)
- [Tips and Tricks and FAQs](HowToFAQ)
- [Index to Dafny Resources for Users](toc)
- [Dafny GitHub project (for developers of the Dafny tools themselves)](https://github.com/dafny-lang/dafny)

![Dafny Banner](./banner.png "Dafny")

> Dafny is a <strong>verification-ready programming language</strong>.

Reviewer 3: ```suggestion
Dafny is a **verification-ready programming language**.
```

As you type in your program, Dafny's verifier constantly looks over your shoulder, flags any errors, shows you counterexamples, and congratulates you when your code matches your specifications.
When you're done, Dafny can <strong>compile your code to C#, Java, JavaScript, Python, C++ or Go</strong> (more to come!), so it can integrate with your existing workflow.

> Dafny will give you <strong>assurance that your code meets the specifications you write</strong>, while letting you write both code and specifications in the Dafny programming language itself.

Reviewer 3: Same as above: `**`, not `<strong>`

Author: Changed
